### PR TITLE
Httprequests

### PR DESCRIPTION
**Pull request by pgruszczynski93**
Pull request after merge

### CODE REVIEW

**Review by PatrykMatwijczuk — APPROVED** [review]

File: PZ/WebClient/Connector.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        self.response = requests.get(self.url_adr, params=self.payload)
+        self.json_data = json.loads(self.response.text)
+
+        for item in self.json_data['resources']:
+            resources_list.append(Resources(resource_id=item['id'], name=item['name'], description=item['description'],
+                                            measurements=item['measurements']))
+
+        page = Page(size=self.json_data['page']['number'], number=self.json_data['page']['totalCount'],
+                    total_count=self.json_data['page']['size'])
+        return [resources_list, page]
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: Remove `[]`, when using comma with return statement, python by default will return tuple

File: PZ/WebClient/Connector.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        self.url_adr = endpoint
+        self.response = requests.get(self.url_adr)
+        self.json_data = json.loads(self.response.text)
+        return Measurements(host=self.json_data['host'], metric=self.json_data['metric'], unit=self.json_data['unit'],
+                            max_value=self.json_data['maxValue'], complex_mes=self.json_data['complex'],
+                            values=self.json_data['values'])
+
+    def get_measurement_values(self, endpoint):
+        values = []
+        self.url_adr = (endpoint+'/values')
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: urls should be created / concatenated with `urljoin`:
http://stackoverflow.com/questions/8223939/how-to-join-absolute-and-relative-urls

Be warry though, in this case `'values'` shouldn't begin with `/

**Review by Dorrin93 — CHANGES_REQUESTED** [review]

File: PZ/WebClient/Connector.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        for item in self.json_data['resources']:
+            resources_list.append(Resources(resource_id=item['id'], name=item['name'], description=item['description'],
+                                            measurements=item['measurements']))
+
+        page = Page(size=self.json_data['page']['number'], number=self.json_data['page']['totalCount'],
+                    total_count=self.json_data['page']['size'])
+        return resources_list, page
+
+    def get_resource_id(self, guid):
+        self.url_adr += guid
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: remove this line

File: PZ/WebClient/Connector.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
+import requests
+import json
+import urllib.parse
+
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: `from urllib.parse import urljoin`

File: PZ/WebClient/Connector.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            resources_list.append(Resources(resource_id=item['id'], name=item['name'], description=item['description'],
+                                            measurements=item['measurements']))
+
+        page = Page(size=self.json_data['page']['number'], number=self.json_data['page']['totalCount'],
+                    total_count=self.json_data['page']['size'])
+        return resources_list, page
+
+    def get_resource_id(self, guid):
+        self.url_adr += guid
+        self.response = requests.get(self.url_adr)
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: `self.response = requests.get(urljoin(self.url_adr, guid))`

File: PZ/WebClient/Connector.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        self.json_data = json.loads(self.response.text)
+        return Resources(resource_id=self.json_data['id'], name=self.json_data['name'], description=self.json_data['description'],
+                         measurements=self.json_data['measurements'])
+
+    def delete_resource(self, guid):
+        self.url_adr += guid
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: remove this line

File: PZ/WebClient/Connector.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+class Connector(object):
+    def __init__(self, url_adr):
+        self.url_adr = url_adr
+        self.response = None
+        self.json_data = None
+        self.payload = {}
+
+    @property
+    def url_address(self):
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: useless, remove

File: PZ/WebClient/Connector.py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        return self.url_adr
+
+    @url_address.setter
+    def url_address(self, url_adr):
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: remove

File: PZ/WebClient/Connector.py [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        self.url_adr = url_adr
+
+    @property
+    def request_args(self):
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: remove

File: PZ/WebClient/Connector.py [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+        return self.payload
+
+    @request_args.setter
+    def request_args(self, payload):
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: rename `request_args` to `payload`

File: PZ/WebClient/Connector.py [hunk 11]
```diff
@@ -0,0 +1,102 @@
+import requests
+import json
+import urllib.parse
+
+from .models import *
+
+class Connector(object):
+    def __init__(self, url_adr):
+        self.url_adr = url_adr
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: add underscore `_` before all members to make them private
`self._url_address = url_address`

File: PZ/WebClient/Connector.py [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+    def get_resource_id(self, guid):
+        self.url_adr += guid
+        self.response = requests.get(self.url_adr)
+        self.json_data = json.loads(self.response.text)
+        return Resources(resource_id=self.json_data['id'], name=self.json_data['name'], description=self.json_data['description'],
+                         measurements=self.json_data['measurements'])
+
+    def delete_resource(self, guid):
+        self.url_adr += guid
+        self.response = requests.delete(self.url_adr)
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: self.response = requests.delete(urljoin(self.url_adr, guid))

File: PZ/WebClient/Connector.py [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+        endpoints_data = json.loads(endpoints)
+        for endpoint in endpoints_data:
+            measurements_list.append(get_measurement(endpoint))
+
+        page = Page(size=self.json_data['page']['number'], number=self.json_data['page']['totalCount'],
+                    total_count=self.json_data['page']['size'])
+        return measurements_list, page
+
+    def get_measurement(self, endpoint):
+        self.url_adr = endpoint
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: remove this line

File: PZ/WebClient/Connector.py [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+        self.response = requests.get(self.url_adr)
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: `self.response = requests.get(endpoint)`

File: PZ/WebClient/Connector.py [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+        self.json_data = json.loads(self.response.text)
+        return Measurements(host=self.json_data['host'], metric=self.json_data['metric'], unit=self.json_data['unit'],
+                            max_value=self.json_data['maxValue'], complex_mes=self.json_data['complex'],
+                            values=self.json_data['values'])
+
+    def get_measurement_values(self, endpoint):
+        values = []
+        self.url_adr += urllib.parse.urljoin(endpoint, 'values')
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: remove this line

File: PZ/WebClient/Connector.py [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+        self.response = requests.get(self.url_adr, params=self.payload)
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: `self.response = requests.get(urljoin(endpoint, 'values'), params=self.payload)`

File: PZ/WebClient/Connector.py [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+        self.json_data = json.loads(self.response.text)
+
+        for item in self.json_data['values']:
+            values.append(Values(value=item['value'], datetime=item['datetime']))
+
+        return values
+
+    def delete_measurement_values(self, guid):
+        self.url_adr += urllib.parse.urljoin(guid, 'values')
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: remove this line

File: PZ/WebClient/Connector.py [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+        self.response = requests.delete(self.url_adr)
+        self.json_data = json.loads(self.response.text)
+        return self.json_data
+
+    def post_measurements(self):
+        self.response = requests.post(self.url_adr, data=json.dumps(self.payload))
+
+    def delete_measurement(self, guid):
+        self.url_adr += urllib.parse.urljoin(guid, 'values')
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: remove this line

File: PZ/WebClient/Connector.py [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+        self.json_data = json.loads(self.response.text)
+
+        for item in self.json_data['values']:
+            values.append(Values(value=item['value'], datetime=item['datetime']))
+
+        return values
+
+    def delete_measurement_values(self, guid):
+        self.url_adr += urllib.parse.urljoin(guid, 'values')
+        self.response = requests.delete(self.url_adr)
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: `self.response = requests.delete(urljoin(self.url_adr, guid+'values'))`

File: PZ/WebClient/Connector.py [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+        self.json_data = json.loads(self.response.text)
+        return self.json_data
+
+    def post_measurements(self):
+        self.response = requests.post(self.url_adr, data=json.dumps(self.payload))
+
+    def delete_measurement(self, guid):
+        self.url_adr += urllib.parse.urljoin(guid, 'values')
+        self.response = requests.delete(self.url_adr)
```
Comment (Dorrin93) [inline PZ/WebClient/Connector.py]: `self.response = requests.delete(urljoin(self.url_adr, guid))`

**Review by Dorrin93 — CHANGES_REQUESTED** [review]

**Review by Dorrin93 — APPROVED** [review]

**Pull request merged**